Fix reparenting after hover delay

AeioMuch opened the pull request:
Fix https://github.com/godotengine/godot/issues/91254
Also keep all nodes selected after reparenting and not just top level ones (before hover delay, because after it's the hovered/inspected node that should be the one selected)

AeioMuch committed:
Fix reparent when hovering after delay

Hilderin commented:
Seems to work perfectly. Thanks!
The only difference with before is that when hovered the rectangle around the node is gone. I think it's brilliant because the rectangle around the node was interfering with the "drop line" making it difficult to see.

ajreckof commented:
One thing that makes me ticks is that the bug was supposedly introduced by #90653 but this fix does not touch code related. This might still be the proper fix but ask the question on why  it does not implicate the code that broke it. I'll try to take a look when I have time most likely sometime between monday and tuesday.

AeioMuch commented:
Hello 👋

> Seems to work perfectly. Thanks! The only difference with before is that when hovered the rectangle around the node is gone. I think it's brilliant because the rectangle around the node was interfering with the "drop line" making it difficult to see.

Thanks for your test and feedback!

> One thing that makes me ticks is that the bug was supposedly introduced by #90653 but this fix does not touch code related. This might still be the proper fix but ask the question on why it does not implicate the code that broke it. I'll try to take a look when I have time most likely sometime between monday and tuesday.

The way I was ensuring that the inspected/hovered Node/TreeItem got selected and added to history when the drag ended was by using `scene_tree->set_selected(node_hovered_now);`, but since the changes in your PR apparently, it look like it update while the drag is happening or something, causing it to break. Just preventing the signal to be emited by passing false `scene_tree->set_selected(node_hovered_now, false);` avoided the issue, but then the problem was that it was not added to the editor selection history anymore after, so I did it at when the drag end instead. I don't think that your PR is problematic per se, just that it changed the previous behaviour, but would be better if you verify indeed.

While I was at it, I removed the use of the `marked` property and directly changed the text color to accent color which visually achieve the same result, with the bonus that it does not lose the accent outline around the target node for the reparent. `set_marked` is forcing the tree to update, causing it to lose the outline, and since now they stay visible I felt it was also better to not set the cursor on TreeItem because that gray outline overlap the other, as pointed by @Hilderin ... 😅

Anyway, if you can take a look at it when you have the time that would be greatly appreciated ! If you think you need to superseed (I think that's the term) it to make some more in depth changes, no problem, or if I need to make more corrections, let me know 👍. Thanks a lot !

AeioMuch commented:
Retested this, and realized this approach still have issue with the path(s) used in the history when reparenting after the delay.

AThousandShips commented:
See also:
* https://github.com/godotengine/godot/pull/91439
* https://github.com/godotengine/godot/pull/91430

KoBeWi reviewed (approved):
The changes look fine. There is a minor issue where the inspected node does not appear at the top, but it can be fixed later:

https://github.com/godotengine/godot/assets/2223172/15d6f372-8a02-46a7-929d-a9c2186483c7

AeioMuch commented:
> The changes look fine. There is a minor issue where the inspected node does not appear at the top, but it can be fixed later:
> 
>  godot.windows.editor.dev.x86_64_HVzun4zCyh.mp4

Yes indeed, right now it will do it only once the drag is ended. Tho I've realized another small issue caused with those new changes is that if you inspected a node by hovering it, then while dragging to the inspector you somehow passed over others node it would not select the inspect node at the end of the drag so I made a small changes that instead of selecting the `node_hovered_now` at the end of the drag, I select the edited object in the inspector (if it can be cast to a Node).

ajreckof commented on editor/scene_tree_dock.cpp:
```diff
@@ -2127,6 +2141,10 @@ void SceneTreeDock::_node_reparent(NodePath p_path, bool p_keep_global_xform) {
 	}
 
 	_do_reparent(new_parent, -1, nodes, p_keep_global_xform);
+
+	for (Node *E : full_selection) {
+		editor_selection->add_node(E);
+	}
```
why is this needed? I'm wondering as this might make #91860 resurface

AeioMuch replied:
In this case it is for consistency so that when you reparent using the right click option `Reparent` it also keep all the reparented nodes selected

ajreckof reviewed (approved):
This looks really good! There is a problem with selection history but this can be fixed later as it is non blocking and only slightly related

akien-mga commented:
Thanks!
